core/src/client/mod.rs

```diff
@@ -399,13 +402,31 @@ pub enum CertificateStore {
 	WebPki,
 }
 
+/// JSON-RPC request object id data type.
+#[derive(Debug, Copy, Clone)]
+pub enum IdKind {
+	/// String.
+	String,
+	/// Number.
+	Number,
+}
+
+impl IdKind {
+	fn into_id(self, id: u64) -> Id<'static> {
+		match self {
+			IdKind::Number => Id::Number(id),
+			IdKind::String => Id::Str(format!("{}", id).into()),
```

Comment:
we could probably use some optimized `itoa` or something but should fast enough I guess for now...

Reply:
Could also subslice byte array of all numbers from `00` to `99` as a static `Cow`, but yeah, this is good enough :).